pkg/meta/tkv.go

```diff
@@ -2598,7 +2756,32 @@ func (m *kvMeta) DumpMeta(w io.Writer) error {
 		dels,
 		tree,
 	}
-	return dm.writeJSON(w)
+	bw, err := dm.writeJsonWithOutTree(w)
+	if err != nil {
+		return err
+	}
+
+	var total int64 = 1 //root
+	progress, bar := utils.NewDynProgressBar("Dump dir progress: ", false)
+	bar.Increment()
+	showProgress := func(totalIncr, currentIncr int64) {
+		total += totalIncr
+		bar.SetTotal(total, false)
+		bar.IncrInt64(currentIncr)
+	}
+	if err = m.dumpDir(m.root, tree, bw, 1, showProgress); err != nil {
+		return err
+	}
+	if _, err = bw.WriteString("\n}\n"); err != nil {
+		return err
+	}
+	if bar.Current() != total {
+		logger.Warnf("Dumped %d / total %d, some entries are not dumped", bar.Current(), total)
+	}
+	bar.SetTotal(0, true)
+	progress.Wait()
+
+	return bw.Flush()
```

Comment:
We should not fail the dump even bw.Flush() failed

Reply:
Doesn't flush failure mean that the result is incomplete?

Reply:
my bad, bw is the writer, never mind.